app/services/reservations/create_at_desk.rb

```diff
@@ -21,6 +21,10 @@ def call
 
     attr_reader :seats, :reservation
 
+    def seats_taken?
+      (Screening.find(@screening_id).all_taken_seats & seats ).empty?
+    end
+
```

Comment:
I think this method name is misleading. You are checking if they are not taken actually. At First look I thought that return condition in call method is wrong but now I see that this method return true if seats are available not taken.

Reply:
Yeah you are absolutely right I've just changed it to seats_available which returns true if he checked not taken yet seat and false if he choose seat that is already taken , thanks a lot for checking it ❤️